### ISSUE TITLE
mobile model has very different inference result

### ISSUE THREAD

**Issue by tensorbuffer** [pytorch/pytorch#116515]
### 🐛 Describe the bug

Step1. Generate the mobile optimized model:
import torch
import torchvision
from torch.utils.mobile_optimizer import optimize_for_mobile

model = torchvision.models.mobilenet_v3_small(pretrained=True)
model.eval()
example = torch.rand(1, 3, 224, 224)
traced_script_module = torch.jit.trace(model, example)
optimized_traced_model = optimize_for_mobile(traced_script_module)
optimized_traced_model._save_for_lite_interpreter("mobilenet_v3_small.ptl")

Step2. Use this model to run an inference, using a husky raw image (attached as zip file)
import torch
import numpy as np
from torch.jit import mobile
mobilenet = mobile._load_for_lite_interpreter("model.mobilenet_v3_small.ptl")

input = np.fromfile("imagenet_husky.raw", dtype=np.float32)
input = input.reshape((1, 3,224,224))
intensor = torch.from_numpy(input)
out = mobilenet(intensor)
_, index = torch.max(out, 1) 

You get index = 574, which does not make sense.

Step3. Use the original mobilenet model to run the same input, and get correct result

from torchvision import models
import torch
import numpy as np
mobilenet = models.mobilenet_v3_small(pretrained=True)
mobilenet.eval()

input = np.fromfile("imagenet_husky.raw", dtype=np.float32)
input = input.reshape((1, 3,224,224))
intensor = torch.from_numpy(input)
out = mobilenet(intensor)
_, index = torch.max(out, 1)

You get index = 250, which is 'Eskimo dog, husky', correct result.

[husky01.zip](https://github.com/pytorch/pytorch/files/13788963/husky01.zip)

### Versions

Collecting environment information...
PyTorch version: 2.0.1+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.10 (default, Jun  4 2021, 15:09:15)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.15.0-88-generic-x86_64-with-glibc2.17
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 2080 Ti
GPU 1: NVIDIA GeForce RTX 2080 Ti

Nvidia driver version: 470.223.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      46 bits physical, 48 bits virtual
CPU(s):                             12
On-line CPU(s) list:                0-11
Thread(s) per core:                 2
Core(s) per socket:                 6
Socket(s):                          1
NUMA node(s):                       1
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              79
Model name:                         Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz
Stepping:                           1
CPU MHz:                            2863.668
CPU max MHz:                        4000.0000
CPU min MHz:                        1200.0000
BogoMIPS:                           7195.26
L1d cache:                          192 KiB
L1i cache:                          192 KiB
L2 cache:                           1.5 MiB
L3 cache:                           15 MiB
NUMA node0 CPU(s):                  0-11
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        KVM: Mitigation: VMX unsupported
Vulnerability L1tf:                 Mitigation; PTE Inversion
Vulnerability Mds:                  Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Meltdown:             Mitigation; PTI
Vulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Mitigation; Clear CPU buffers; SMT vulnerable
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a rdseed adx smap intel_pt xsaveopt cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts md_clear flush_l1d

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] torch==2.0.1+cpu
[pip3] torchaudio==2.0.2+cpu
[pip3] torchvision==0.15.2+cpu
[conda] numpy                     1.24.1                   pypi_0    pypi
[conda] torch                     2.0.1+cpu                pypi_0    pypi
[conda] torchaudio                2.0.2+cpu                pypi_0    pypi
[conda] torchvision               0.15.2+cpu               pypi_0    pypi

**Comment by ZhaoqiongZ** [comment]
The husky01.zip file is broken, however I use another husky picture and can get the same conclusion that the inference result turns wrong for mobile optimized model.

**Comment by tensorbuffer** [comment]
ok thanks for checking!
I can download the zip file and unzip and it's same as what I use.
We can only use this version of pytorch, I think this is the last one that still uses C++14, the next release uses c++17, which we can't upgrade easily (relates to android NDK/SDK limitation). So when the fix come I hope it can be applied to this version of pytorch.

**Comment by jideilori** [comment]
Yup, facing the same issue

**Comment by thelou1s** [comment]
I have a similar but a little different problem, the model have good accuracy on python but bad accuracy in android.
I'm confused, what may be the real problem?

![image](https://github.com/pytorch/pytorch/assets/5654770/eb0b9f4d-7b6a-403c-ad94-112f0e805372)

**Comment by tensorbuffer** [comment]
@albanD any update? pytorch mobile is not supported any more?

**Comment by albanD** [comment]
I'm afraid not :( the efforts of that team have been refocused on https://pytorch.org/executorch-overview which should be a much more realiable solution!

**Comment by AkiSakurai** [comment]
I've observed that the bias is lost during the batch normalization convolution fusion optimization. Here's a proposed fix.
https://github.com/pytorch/pytorch/compare/main...AkiSakurai:pytorch:optimize_for_mobile_missing_bias
I'm not entirely confident that this fix works in every scenario, as I'm not well-versed in the codebase and I lack the time to polish it. Perhaps someone else interested could complete the task.
